src/core/settings.h

```diff
@@ -4,24 +4,97 @@
     Licensed under the MIT License.
 
 --*/
+typedef struct QUIC_SETTINGS_INTERNAL {
```

Comment:
How are we going to deal with the problem of this getting out of sync with the public type? I don't like the idea of a copy + more... Need to figure out a better way.

Reply:
You just have to manually keep them in sync? I have a static assert to keep them in sync.

Reply:
We could have an internal type that just contains the public type as an anonymous member and adds additional fields?
But then the internal fields would have to have public flags. Or a separate `InternalFlags` field. 
What do you think of this?
```c
typedef struct QUIC_SETTINGS_INTERNAL {
    QUIC_SETTINGS; // Not all compilers may support this
    union {
        uint64_t InternalSetFlags;
        struct {
            uint64_t GeneratedCompatabilitylist :1;
        } InternalSet;
    }
    uint32_t* GeneratedCompatabilityList;
    uint32_t GeneratedCompatabililtyListLength;
} QUIC_SETTINGS_INTERNAL;
```

Reply:
Yeah, I was also considering this design. Thoughts?

Reply:
Struct packing will be worse than the current copy&add, since the struct will have 8-byte alignment, and as fields are added (like the new fields I just added) it will have some padding between it and the internal-only fields. We can work to minimize that as much as possible, but the possibility that some padding is present is unavoidable.

Reply:
C++ doesn't allow anonymous embedded struct. (So the unit tests and the attack tool don't build because they include the internal headers directly.)

Reply:
Is there a reason it needs to be anonymous? Thats the only part c++ doesn't like iirc.

Reply:
There are ways to make C++ happy (if C++, inherit; if C, anonymous).

Reply:
I'm just going to remove this.